### PR TITLE
feat: File backed variable values

### PR DESCRIPTION
**Pull request by disq**

### CODE REVIEW

File: docs/configuration/environment-variable-substitution.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      connection {
+        dsn = "${DSN}"
+      }
+    }
+    
+    provider "aws" {
+      configuration {
+        accounts "<YOUR ID>" {
+          role_arn = "${ARN}"
+        }
```
Comment (bbernays) [inline docs/configuration/environment-variable-substitution.md:26]: Does it work if you have multiple accounts?

```suggestion
        accounts "012345678910" {
          role_arn = "${ARN_012345678910}"
        }
        accounts "012345678911" {
          role_arn = "${ARN_012345678911}"
        }

```

File: docs/configuration/environment-variable-substitution.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    cloudquery {
+      provider "aws" {
+        source  = ""
+        version = "${AWS_VERSION}"
+      }
+
+      connection {
+        dsn = "${DSN}"
+      }
```
Comment (bbernays) [inline docs/configuration/environment-variable-substitution.md:19]: Could a customer combine env variables together?

```suggestion
      connection {
        dsn = "postgres://${USERNAME}:${PASSWORD}@${HOST}:${PORT}/${DBNAME}?sslmode=${SSLMODE}"
      }
```

File: docs/configuration/environment-variable-substitution.md (hunk 1 above)
Comment (disq) [reply docs/configuration/environment-variable-substitution.md:26]: I hope so :)

File: docs/configuration/environment-variable-substitution.md (hunk 2 above)
Comment (disq) [reply docs/configuration/environment-variable-substitution.md:19]: Yes I think

**Review by bbernays — APPROVED** [review]
LGTM

File: docs/configuration/file-backed-variable-values.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+It's possible to read a configuration value from the filesystem using the `file()` function. This function reads the contents of the file logically and sets the contents directly to the related variable:
+
+    cloudquery {
+      provider "aws" {
+        // ...
+      }
+
+      connection {
+            dsn = file("dsn.txt")
```
Comment (ukmadlz) [inline docs/configuration/file-backed-variable-values.md:11]: Just a little indentation fix

**Review by ukmadlz — CHANGES_REQUESTED** [review]

File: docs/configuration/file-backed-variable-values.md (hunk 3 above)
Comment (disq) [reply docs/configuration/file-backed-variable-values.md]: indented

**Review by ukmadlz — APPROVED** [review]

**Pull request merged**